ts/screens/wallet/AddCardScreen.tsx

```diff
@@ -95,301 +98,201 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
-          leftButton={secondaryButtonProps}
-          rightButton={primaryButtonPropsFromState(this.state)}
-        />
-      </BaseScreenComponent>
-    );
-  }
-
-  private isValidPan() {
-    return this.state.pan.map(pan => CreditCardPan.is(pan)).toUndefined();
-  }
-
-  private isValidExpirationDate() {
-    return this.state.expirationDate
-      .map(expirationDate => {
-        const [expirationMonth, expirationYear] = expirationDate.split("/");
-        return (
-          CreditCardExpirationMonth.is(expirationMonth) &&
-          CreditCardExpirationYear.is(expirationYear) &&
-          !isExpired(Number(expirationMonth), Number(expirationYear))
-        );
-      })
-      .toUndefined();
-  }
-
-  private isValidSecurityCode() {
-    return this.state.securityCode
-      .map(securityCode => CreditCardCVC.is(securityCode))
-      .toUndefined();
-  }
-
-  private updateHolderState(value: string) {
-    this.setState({
-      holder: value !== EMPTY_CARD_HOLDER ? some(value) : none
-    });
-  }
-
-  private updatePanState(value: string) {
-    this.setState({
-      pan: value && value !== EMPTY_CARD_PAN ? some(value) : none
-    });
-  }
-
-  private updateExpirationDateState(value: string) {
-    this.setState({
-      expirationDate:
-        value && value !== EMPTY_CARD_EXPIRATION_DATE ? some(value) : none
-    });
-  }
+                      ? "wallet.dummyCard.values.securityCode4D"
+                      : "wallet.dummyCard.values.securityCode"
+                  ),
+                  returnKeyType: "done",
+                  maxLength: 4,
+                  type: "custom",
+                  options: { mask: "9999" },
+                  keyboardType: "numeric",
+                  secureTextEntry: true,
+                  includeRawValueInChangeText: true,
+                  onChangeText: value => updateState("securityCode", value)
+                }}
+              />
+            </Col>
+          </Grid>
+
+          <View spacer={true} />
+
+          <Link onPress={present}>{I18n.t("wallet.missingDataCTA")}</Link>
+
+          <View spacer />
+
+          <Link onPress={openSupportedCardsPage}>
+            {I18n.t("wallet.openAcceptedCardsPageCTA")}
+          </Link>
+        </Content>
+      </ScrollView>
+      <SectionStatusComponent sectionKey={"credit_card"} />
+      <FooterWithButtons
+        type="TwoButtonsInlineHalf"
+        leftButton={props.secondaryButtonProps}
+        rightButton={primaryButtonPropsFromState(
+          state,
+          props.navigateToConfirmCardDetailsScreen
+        )}
+      />
+    </BaseScreenComponent>
+  );
+};
 
-  private updateSecurityCodeState(value: string) {
-    this.setState({
-      securityCode:
-        value && value !== EMPTY_CARD_SECURITY_CODE ? some(value) : none
-    });
+const mapStateToProps = (state: GlobalState, props: OwnProps) => ({
+  profileNameSurname: profileNameSurnameSelector(state),
+  secondaryButtonProps: {
```

Comment:
Since this property doesn't access any store values it could be placed inside the component as const.
Furthermore it is never used

Reply:
Doesn't it use `props.navigation.goBack()`?

Reply:
@DavideValdo it access a prop that can be passed through a `mapDispatchToProps`, I would suggest to move the _goBack_ function on it dispatching a `navigateBack` action